shared/metrics/src/test/java/io/pravega/shared/metrics/StatsLoggerProxyTest.java

```diff
@@ -67,23 +67,20 @@ public void testGetOrCreate() {
         createMetrics(proxy, "", "hostname", "localhost");
         Assert.assertEquals("Unexpected number of metrics registered initially.", METRIC_COUNT * 2, metrics.size());
 
-        // Re-create/request the same metrics. Verify the original ones have not been touched, but we did create (and close)
-        // a new set.
+        // Re-create/request the same metrics. Verify the original ones have not been touched.
         createMetrics(proxy, "");
-        Assert.assertEquals("Unexpected number of metrics registered after re-registering same names.", 3 * METRIC_COUNT, metrics.size());
+        Assert.assertEquals("Unexpected number of metrics registered after re-registering same names.", 2 * METRIC_COUNT, metrics.size());
```

Comment:
The `expected` argument has been changed because before the expectation that was we would still create the `Metric` and add it to the `metrics` array, but just immediately mark it as closed. Now we just avoid that creation step completely (if it exists).